### PR TITLE
updates for 0.12, remove foreign-generics

### PR DESCRIPTION
**Pull request by justinwoo**
Foreign-Generics won't be updated for a while, but the usages here seem to be leftover from pre-11.6 days. I updated this to have the user supply their own decoding function, and updated the tests to use Simple-JSON.

### CODE REVIEW

**Comment by anttih** [comment]
Thanks! I really would like to use foreign-generics, was the hold-up just with some record stuff? Do you think we should make an interim release with this change and then update to foreign-generics later?

**Comment by justinwoo** [comment]
I would like to use this library without foreign-generics at all though. Those who want to use foreign-generics could later opt in when it gets updated.

**Comment by anttih** [comment]
How would they opt in later if this library doesn't use foreign-generic anymore?

I'm skeptical of removing the `Decode` constraint which would completely change the way this library works. Are you using simple-json instead of foreign-generic? Couldn't you write a `Decode` instance that uses simple-json or not decode at all but decode manually the `Foreign` value you get? Is there no type class that we could use that would serve everyone?

**Comment by justinwoo** [comment]
They can always recover the constraint version by binding show and ForeignGeneric.decode.

I would never want to use Decode in my code base which purposefully doesn't use Decode for anything. Decoding the Foreign value I get back is exactly what this PR does, where no assumptions are made but for the user to provide a function to Either.

I think there is absolutely no type class that would serve everyone, and nobody really should expect one to work. This library should not force a specific decoding solution to its users.

Libraries forcing very specific unrelated typeclasses is what makes them impossible to actually use with any of my other tools, and, for example, is why I don't use the purescript-express package.

I wouldn't be that opposed to having a fork of this library called "node-postgres-core" and having this one maintain some Foreign-Generics constraints if the README at least pointed people at not using Decode for everything.

**Comment by anttih** [comment]
One option that converges with the `node-postgres-core` package you suggested, would be to only decode as much as `node-pg` does. That is, we would provide a custom `FromSql` class that would provide an instance basically only for strings, `Foreign` and records using `RowToList`. That way you would get at least something out from the result rows, and the API would be much cleaner.

**Comment by justinwoo** [comment]
I really don't want this library to do *any* actual decoding on my behalf, I want to pass in all the decoding I do with concrete functions.

**Comment by anttih** [comment]
Could we get rid of the `print` argument by fixing the error type to `Error`, which can be used with `Aff`?

File: src/Database/Postgres.purs [hunk 1]
```diff
@@ -90,93 +85,87 @@ connectionInfoFromConfig c p = unsafeCoerce
...
 
 -- | Just like `query` but does not make any param replacement
-query_ :: forall eff a
-  . Decode a
-  => Query a -> Client -> Aff (db :: DB | eff) (Array a)
-query_ (Query sql) client = do
-  rows <- fromEffFnAff $ runQuery_ sql client
-  either liftError pure (runExcept (sequence $ decode <$> rows))
+query_ :: forall b a
+  . (b -> String) -> (Foreign -> Either b a) ->  Query a -> Client -> Aff (Array a)
```
Comment (anttih) [inline src/Database/Postgres.purs]: There is an extra space before the `Query a` here and also in the next few type defs.

**Comment by justinwoo** [comment]
All right, updated to use Error

**Comment by anttih** [comment]
One final nitpick, the commit still says `Either b a`. Could you fix that?

**Comment by justinwoo** [comment]
okay, i updated it

**Comment by anttih** [comment]
Thanks!

**Pull request merged**